datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -944,6 +959,100 @@ private void handleLeaderDoAssignment() {
     }
   }
 
+  /**
+   * assign the partition to tasks for a particular datastreamGroup
+   *
+   * @param maybeDatastreamGroupName the datastreamGroup that needs to perform the partition assignment
+   */
+  private void performPartitionAssignment(Optional<String> maybeDatastreamGroupName) {
+    if (!maybeDatastreamGroupName.isPresent()) {
+      _log.error("Datastream group is not found when performing partition assignment");
+      return;
+    }
+
+    boolean succeeded = false;
+    Map<String, Set<DatastreamTask>> previousAssignmentByInstance = new HashMap<>();
```

Comment:
It's checked when we mutating the task. The task mutation will fail and throw an DatastreamTransientException so that we can retry. I do it in this way so that we can minimize the task checking to zk.